NMS-15859: Inherit basic SSL config props from the producer to the stream client. (#6412)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/kafka-producer/configure-kafka.adoc

----

In the example above, we set the vertex and edge topics to be different by default.

> 

Reviewer: ```suggestion
```

== Configure the alarm sync streams client

> When producing alarms to a topic, we also automatically enable a Kafka Stream client to help synchronize the contents of the topic and ensure that it remains eventually consistent with the database.

Reviewer 2: ```suggestion
When producing alarms to a topic, we also automatically enable a Kafka Stream client to help synchronize the contents of the topic and ensure that it eventually remains consistent with the database.
```
For my own understanding, does this mean it takes noticeable time to synchronize?

Author: Yeah, there's some lag, and it will get there, eventually. 

Given that "[eventual consistency](https://en.wikipedia.org/wiki/Eventual_consistency)" is a known term, should these stay together?

Reviewer 2: Ah, should have checked. In that case I'd suggest:

```suggestion
When producing alarms to a topic, we also automatically enable a Kafka Stream client to help synchronize the contents of the topic and ensure eventual consistency with the database.
```

> The streams client takes different properties than the producer and requires a separate configuration map.

Reviewer: "streams client" or "stream client"? it's singular when mentioned in the previous sentence. (and captialized)

> We automatically pull known properties from the producer configuration, allowing it to work without further configuration, in most cases.

Reviewer 2: ```suggestion
We automatically pull known properties from the producer configuration, allowing it to work without further configuration in most cases.
```

> If your producer takes special configuration directive, or you would like to tune the behavior of the stream client, you can set properties in `org.opennms.features.kafka.producer.streams`:

Reviewer 2: ```suggestion
If your producer takes a special configuration directive, or you would like to tune the behavior of the stream client, you can set properties in `org.opennms.features.kafka.producer.streams`:
```
